### ISSUE TITLE
RangeError was thrown: Value not in range

### ISSUE THREAD

**Issue by certaintls** [Ephenodrom/Dart-Basic-Utils#9]
When I was decoding this certificate, I ran into the error below:

````
[ +806 ms] I/flutter (28085): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
[        ] I/flutter (28085): The following RangeError was thrown building:
[        ] I/flutter (28085): Value not in range: 6090798525784719588
[        ] I/flutter (28085): 
[        ] I/flutter (28085): When the exception was thrown, this was the stack:
[        ] I/flutter (28085): #0      _rangeCheck (dart:typed_data-patch/typed_data_patch.dart:4791:5)
[        ] I/flutter (28085): #1      _ByteBuffer.asUint8List (dart:typed_data-patch/typed_data_patch.dart:1931:5)
[        ] I/flutter (28085): #2      new Uint8List.view (dart:typed_data:962:19)
[        ] I/flutter (28085): #3      ASN1Parser.nextObject (package:asn1lib/asn1parser.dart:45:30)
[        ] I/flutter (28085): #4      X509Utils.x509CertificateFromPem (package:basic_utils/src/X509Utils.dart:328:33)
````

The PEM string is below:

````
-----BEGIN CERTIFICATE-----
MIIClDCCAhqgAwIBAgIILCmcWxbtBZUwCgYIKoZIzj0EAwIwfzELMAkGA1UEBhMC
VVMxDjAMBgNVBAgMBVRleGFzMRAwDgYDVQQHDAdIb3VzdG9uMRgwFgYDVQQKDA9T
U0wgQ29ycG9yYXRpb24xNDAyBgNVBAMMK1NTTC5jb20gRVYgUm9vdCBDZXJ0aWZp
Y2F0aW9uIEF1dGhvcml0eSBFQ0MwHhcNMTYwMjEyMTgxNTIzWhcNNDEwMjEyMTgx
NTIzWjB/MQswCQYDVQQGEwJVUzEOMAwGA1UECAwFVGV4YXMxEDAOBgNVBAcMB0hv
dXN0b24xGDAWBgNVBAoMD1NTTCBDb3Jwb3JhdGlvbjE0MDIGA1UEAwwrU1NMLmNv
bSBFViBSb290IENlcnRpZmljYXRpb24gQXV0aG9yaXR5IEVDQzB2MBAGByqGSM49
AgEGBSuBBAAiA2IABKoSR5CYG/vvw0AHgyBO8TCCogbR8pKGYfL2IWjKAMTH6kMA
VIbc/R/fALhBYlzccBYy3h+Z1MzFB8gIH2EWB1E9fVwHU+M1OIzfzZ/ZLg1Kthku
WnBaBu2+8KGwytAJKaNjMGEwHQYDVR0OBBYEFFvKXuXe0oGqzagtZFG22XKbl+ZP
MA8GA1UdEwEB/wQFMAMBAf8wHwYDVR0jBBgwFoAUW8pe5d7SgarNqC1kUbbZcpuX
5k8wDgYDVR0PAQH/BAQDAgGGMAoGCCqGSM49BAMCA2gAMGUCMQCK5kCJN+vp1RPZ
ytRrJPOwPYdGWBrssd9v+1a6cGvHOMzosYxPD/fxZ3YOg9AeUY8CMD32IygmTMZg
h5Mmm7I1HrrW9zzRHM76JTymGoEVW/MSD2zuZYrJh6j5B+BimoxcSg==
-----END CERTIFICATE-----
````

The data in text format is:
````
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 3182246526754555285 (0x2c299c5b16ed0595)
    Signature Algorithm: ecdsa-with-SHA256
        Issuer: C=US, ST=Texas, L=Houston, O=SSL Corporation, CN=SSL.com EV Root Certification Authority ECC
        Validity
            Not Before: Feb 12 18:15:23 2016 GMT
            Not After : Feb 12 18:15:23 2041 GMT
        Subject: C=US, ST=Texas, L=Houston, O=SSL Corporation, CN=SSL.com EV Root Certification Authority ECC
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
                pub:
                    04:aa:12:47:90:98:1b:fb:ef:c3:40:07:83:20:4e:
                    f1:30:82:a2:06:d1:f2:92:86:61:f2:f6:21:68:ca:
                    00:c4:c7:ea:43:00:54:86:dc:fd:1f:df:00:b8:41:
                    62:5c:dc:70:16:32:de:1f:99:d4:cc:c5:07:c8:08:
                    1f:61:16:07:51:3d:7d:5c:07:53:e3:35:38:8c:df:
                    cd:9f:d9:2e:0d:4a:b6:19:2e:5a:70:5a:06:ed:be:
                    f0:a1:b0:ca:d0:09:29
                ASN1 OID: secp384r1
                NIST CURVE: P-384
        X509v3 extensions:
            X509v3 Subject Key Identifier: 
                5B:CA:5E:E5:DE:D2:81:AA:CD:A8:2D:64:51:B6:D9:72:9B:97:E6:4F
            X509v3 Basic Constraints: critical
                CA:TRUE
            X509v3 Authority Key Identifier: 
                keyid:5B:CA:5E:E5:DE:D2:81:AA:CD:A8:2D:64:51:B6:D9:72:9B:97:E6:4F

            X509v3 Key Usage: critical
                Digital Signature, Certificate Sign, CRL Sign
    Signature Algorithm: ecdsa-with-SHA256
         30:65:02:31:00:8a:e6:40:89:37:eb:e9:d5:13:d9:ca:d4:6b:
         24:f3:b0:3d:87:46:58:1a:ec:b1:df:6f:fb:56:ba:70:6b:c7:
         38:cc:e8:b1:8c:4f:0f:f7:f1:67:76:0e:83:d0:1e:51:8f:02:
         30:3d:f6:23:28:26:4c:c6:60:87:93:26:9b:b2:35:1e:ba:d6:
         f7:3c:d1:1c:ce:fa:25:3c:a6:1a:81:15:5b:f3:12:0f:6c:ee:
         65:8a:c9:87:a8:f9:07:e0:62:9a:8c:5c:4a
SHA1 Fingerprint=4C:DD:51:A3:D1:F5:20:32:14:B0:C6:C5:32:23:03:91:C7:46:42:6D
````

ASN.1 JavaScript decoder of this certificate is at http://lapo.it/asn1js/#MIIClDCCAhqgAwIBAgIILCmcWxbtBZUwCgYIKoZIzj0EAwIwfzELMAkGA1UEBhMCVVMxDjAMBgNVBAgMBVRleGFzMRAwDgYDVQQHDAdIb3VzdG9uMRgwFgYDVQQKDA9TU0wgQ29ycG9yYXRpb24xNDAyBgNVBAMMK1NTTC5jb20gRVYgUm9vdCBDZXJ0aWZpY2F0aW9uIEF1dGhvcml0eSBFQ0MwHhcNMTYwMjEyMTgxNTIzWhcNNDEwMjEyMTgxNTIzWjB_MQswCQYDVQQGEwJVUzEOMAwGA1UECAwFVGV4YXMxEDAOBgNVBAcMB0hvdXN0b24xGDAWBgNVBAoMD1NTTCBDb3Jwb3JhdGlvbjE0MDIGA1UEAwwrU1NMLmNvbSBFViBSb290IENlcnRpZmljYXRpb24gQXV0aG9yaXR5IEVDQzB2MBAGByqGSM49AgEGBSuBBAAiA2IABKoSR5CYG_vvw0AHgyBO8TCCogbR8pKGYfL2IWjKAMTH6kMAVIbc_R_fALhBYlzccBYy3h-Z1MzFB8gIH2EWB1E9fVwHU-M1OIzfzZ_ZLg1KthkuWnBaBu2-8KGwytAJKaNjMGEwHQYDVR0OBBYEFFvKXuXe0oGqzagtZFG22XKbl-ZPMA8GA1UdEwEB_wQFMAMBAf8wHwYDVR0jBBgwFoAUW8pe5d7SgarNqC1kUbbZcpuX5k8wDgYDVR0PAQH_BAQDAgGGMAoGCCqGSM49BAMCA2gAMGUCMQCK5kCJN-vp1RPZytRrJPOwPYdGWBrssd9v-1a6cGvHOMzosYxPD_fxZ3YOg9AeUY8CMD32IygmTMZgh5Mmm7I1HrrW9zzRHM76JTymGoEVW_MSD2zuZYrJh6j5B-BimoxcSg

Not sure how to proceed to debug, if you can point a direction or have a suggestion, I would appreciate!

**Comment by Ephenodrom** [comment]
Hello @certaintls ,
Thanks for the issue, this seems to be a bug in the ASN1 Package I use in my package. Will look into it tomorrow and try to fix it in the ASN1 Package ( https://github.com/wstrange/asn1lib ).

Will give feedback when I have more information.

Regards

**Comment by Ephenodrom** [comment]
Hello @certaintls 
I found the bug. The error comes from parsing the public key information. I fixed it with version 2.4.9. Please update and give me some feedback :).

Please use the package to test more PEM String, due to I could not test all possible certificates, there might still be some other problems. There are also a lot of ObjectIdentifier missing in the ASN1 package. The OID "ecdsa-with-SHA256" for example is not recognized by the ASN1 Package. Feel free to create an issue for any bug you find or any missing oid. Will try to add it to the ASN1 package.

Thanks for using this package and I hope you can recommend it to other developers.

Regards

**Comment by certaintls** [comment]
@Ephenodrom Thanks a lot for the fix! I can confirm it is working. I opened another one https://github.com/Ephenodrom/Dart-Basic-Utils/issues/10 although, I am not sure if it belongs to upstream or this project.
